tests/test_Sequence.cc

```diff
@@ -2967,6 +2968,77 @@ TEST_CASE("execute_sequence(): Messages", "[execute_sequence]")
     REQUIRE(msg.get_timestamp() - t0 < 1s);
 }
 
+TEST_CASE("execute_sequence(): Message callbacks", "[execute_sequence]")
+{
+    std::string str;
+
+    Context context;
+    context.message_callback_function = [&str](const Message& msg)
```

Comment:
Fair enough, I'll add it. I am not a huge fan of adding return types to lambdas where they are not needed because that adds more visual noise, but I understand why this might be desirable.

Reply:
Done.